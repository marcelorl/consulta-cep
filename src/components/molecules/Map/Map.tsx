import React from 'react'
import styled from 'styled-components'
import GoogleMapReact, { Coords } from 'google-map-react'

import markerSvg from './images/marker.svg'

interface IMap {
  coordinates?: Coords
}

const MapWrapper = styled.div`
  height: 100vh;
  width: 100%;
`

const MarkerWrapper = styled.div`
  height: 2rem;
  width: 2rem;
`

const Marker = React.memo((props: any) =>
  <MarkerWrapper>
    {props.children}
  </MarkerWrapper>
)

const Map = (props: IMap) =>
  <MapWrapper>
    <GoogleMapReact
      bootstrapURLKeys={{ key: process.env.REACT_APP_GOOGLE_MAPS_API_KEY }}
      defaultCenter={{lat: -15.5, lng: -57}}
      center={props.coordinates}
      defaultZoom={15}
    >
      <Marker {...props.coordinates}>
        <img src={markerSvg} alt="marker"/>
      </Marker>
    </GoogleMapReact>
  </MapWrapper>

export default Map
