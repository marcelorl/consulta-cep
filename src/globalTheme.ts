import { DefaultTheme } from 'styled-components'

export const globalTheme: DefaultTheme = {
  colors: {
    darkgray: '#999999',
    gainsboro: '#e0ded9',
    lightblue: '#aadaff',
    white: '#ffffff',
  }
}
