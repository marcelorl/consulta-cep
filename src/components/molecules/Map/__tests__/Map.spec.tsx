import React from 'react'
import { ThemeProvider } from 'styled-components'
import { cleanup, render } from 'react-testing-library'

import Map from '../Map'
import { globalTheme } from '../../../../globalTheme'

afterEach(cleanup)

describe('#Map', () => {
  it('should render Map', () => {
    const { container } = render(
      <ThemeProvider theme={globalTheme}>
        <Map />
      </ThemeProvider>
    )

    expect(container).toMatchSnapshot()
  })
})


