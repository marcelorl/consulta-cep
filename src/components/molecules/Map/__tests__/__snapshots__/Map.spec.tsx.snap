// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Map should render Map 1`] = `
.c0 {
  height: 100vh;
  width: 100%;
}

.c1 {
  height: 2rem;
  width: 2rem;
}

<div>
  <div
    class="c0"
  >
    <div
      style="width: 100%; height: 100%; margin: 0px; padding: 0px; position: relative;"
    >
      <div
        style="width: 100%; height: 100%; left: 0px; top: 0px; margin: 0px; padding: 0px; position: absolute;"
      />
      <div
        style="width: 50%; height: 50%; left: 50%; top: 50%; margin: 0px; padding: 0px; position: absolute;"
      >
        <div
          style="width: 100%; height: 100%; left: 0px; top: 0px; margin: 0px; padding: 0px; position: absolute;"
        >
          <div
            class="c1"
          >
            <img
              alt="marker"
              src="marker.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
